customer-service/src/main/kotlin/cz/kss/proj/clientservice/repository/CustomerRepositoryTest.kt

```diff
@@ -0,0 +1,109 @@
+    private val dbClient: DatabaseClient,
+    private val addressRepositoryTest: AddressRepositoryTest,
+    private val additionalInformationRepositoryTest: AdditionalInformationRepositoryTest,
+) : CustomerRepositoryT {
+    override fun findAll(): Flow<Customer> {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun deleteAll(): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun entity(f: () -> DatabaseClient.GenericExecuteSpec): Customer? {
+        return super.entity(f)
+    }
+
+    override fun entities(f: () -> Flow<MutableMap<String, Any>>): Flow<Customer> {
+        return super.entities(f)
+    }
+
+    override fun toEntity(row: MutableMap<String, Any>): Customer? {
+        return if (
+            (row["id"] == null || row["id"] !is Long) ||
+            (row["user_name"] == null) ||
+            (row["email"] == null)
+        ) {
+            null
+        } else {
+            Customer(
+                id = row["id"] as Long,
+                userName = row["user_name"] as String,
+                firstName = row["first_name"] as String,
+                lastName = row["last_name"] as String,
+                email = row["email"] as String,
+                additionalInformation = null,
+                address = null
+            )
+        }
+    }
+
+    override suspend fun delete(entity: Customer): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun save(entity: Customer): Customer {
+        return saveAdditionalInformation(saveAddress(this.saveCustomer(entity)))
+    }
+
+    suspend fun saveCustomer(entity: Customer): Customer {
+        return entity.id?.let { id ->
+            dbClient.sql(
+                """
+                        UPDATE customer SET user_name = :user_name,first_name = :first_name,last_name = :last_name, email = :email WHERE id = :id
+                    """.trimIndent()
+            )
+                .bind("user_name", entity.userName)
+                .bind("first_name", entity.firstName)
+                .bind("last_name", entity.lastName)
+                .bind("email", entity.email)
+                .fetch().first()
+                .mapNotNull { toEntity(it) }
+                .awaitSingle()
+        } ?: run {
+            dbClient.sql(
+                """
+                        INSERT INTO customer (user_name, first_name,last_name,email) VALUES (:user_name,:first_name,:last_name, :email)
+                    """.trimIndent()
+            )
+                .bind("user_name", entity.userName)
+                .bind("first_name", entity.firstName)
+                .bind("last_name", entity.lastName)
+                .bind("email", entity.email)
+                .filter { statement -> statement.returnGeneratedValues("id") }
+                .fetch().one()
+                .doOnNext { entity.id = it["id"] as Long }
+                .thenReturn(entity)
+                .awaitSingle()
```

Comment:
Please look at this @zezutom, I have to remove entity dsl because there was some problem with current implementation, i replace your impl for this as you can see there: https://neilwhite.ca/joins-with-spring-data-r2dbc/ , but I am not sure if it is good approach. I will try to find another impl but I am not sure if some else approach or better approach exists